model_analyzer/result/result_table_manager.py

```diff
@@ -0,0 +1,609 @@
+                           ignore_widths=True,
+                           include_title=False)
+
+    def _export_model_csv(self, inference_writer, gpu_metrics_writer,
+                          column_separator):
+        """
+        Writes the model table as a csv file using the given writer
+
+        Parameters
+        ----------
+        inference_writer : OutputWriter
+            Used to write the inference table result to an output stream
+        gpu_metrics_writer : OutputWriter
+            Used to write the gpu metrics table result to an output stream
+        column_separator : str
+            The string that will be inserted between each column
+            of the table
+
+        Raises
+        ------
+        TritonModelAnalyzerException
+        """
+
+        gpu_table = self._result_tables[self.model_gpu_table_key]
+        self._write_result(table=gpu_table,
+                           writer=gpu_metrics_writer,
+                           column_separator=column_separator,
+                           ignore_widths=True,
+                           include_title=False)
+
+        non_gpu_table = self._result_tables[self.model_inference_table_key]
+        self._write_result(table=non_gpu_table,
+                           writer=inference_writer,
+                           column_separator=column_separator,
+                           ignore_widths=True,
+                           include_title=False)
+
+    def _write_results(self, writer, column_separator):
+        """
+        Writes the tables using the writer with the given column
+        specifications.
+
+        Parameters
+        ----------
+        writer : OutputWriter
+            Used to write the result tables to an output stream
+        column_separator : str
+            The string that will be inserted between each column
+            of the table
+
+        Raises
+        ------
+        TritonModelAnalyzerException
+        """
+
+        for table in self._result_tables.values():
+            self._write_result(table,
+                               writer,
+                               column_separator,
+                               ignore_widths=False)
+
+    def _write_result(self,
+                      table,
+                      writer,
+                      column_separator,
+                      ignore_widths=False,
+                      include_title=True):
+        """
+        Utility function that writes any table
+        """
+
+        if include_title:
+            writer.write('\n'.join([
+                table.title() + ":",
+                table.to_formatted_string(separator=column_separator,
+                                          ignore_widths=ignore_widths), "\n"
+            ]))
+        else:
+            writer.write(
+                table.to_formatted_string(separator=column_separator,
+                                          ignore_widths=ignore_widths) + "\n\n")
```

Comment:
Is the extra new line intentional here?  
To refactor this method slightly, I would build the string and then make 1 call to the writer with the formatted string.

Reply:
Thats fine. This would be a boyscout change at most then. Not necessary for this story.